Windows Headers Not Included

gsawin opened the pull request:
To successfully compile on Windows, XMLWriter.m was modified to include Windows platform headers by adding #include "PLT/BoxPLTPlatformTarget.h".

gsawin committed:
Windows Headers Not Included

To successfully compile on Windows, XMLWriter.m was modified to include Windows platform headers by adding #include "PLT/BoxPLTPlatformTarget.h".

ChrisJEChandler reviewed (approved):
Looks good to me. The only thing here is that XMLWriter is external source so we might want to put some kind of #if () condition around the inclusion of BoxPLTPlatformTarget.h so that it's included only if building AmiShare.
EG
#if (AMISHARE_USING_GCC || AMISHARE_USING_CLANG)
#include "PLT/BoxPLTPlatformTarget.h:
#endif /* (AMISHARE_USING_GCC || AMISHARE_USING_CLANG) */
The alternative to this is to embed the essential definitions in BoxPLTPlatformTarget.h in XMLWriter.m, but this is a pain.

gsawin committed:
Add Preprocessor Directive

A conditional preprocessor directive was added to prevent inclusion of the PLT/BoxPLTPlatformTarget.h header file in non Kinnami AmiShare builds.
